applications/MeshingApplication/custom_strategies/custom_convergencecriterias/error_mesh_criteria.h

```diff
@@ -0,0 +1,481 @@
+                "filename"                             : "out",
+                "framework"                            : "Lagrangian",
+                "internal_variables_parameters"        :
+                {
+                    "allocation_size"                      : 1000, 
+                    "bucket_size"                          : 4, 
+                    "search_factor"                        : 2, 
+                    "interpolation_type"                   : "LST",
+                    "internal_variable_interpolation_list" :[]
+                },
+                "save_external_files"              : false,
+                "max_number_of_searchs"            : 1000,
+                "echo_level"                       : 3
+            },
+            "error_strategy_parameters": 
+            {
+                "minimal_size"                        : 0.1,
+                "maximal_size"                        : 10.0, 
+                "enforce_current"                     : true
+            }
+        })" );
+        
+        mThisParameters.ValidateAndAssignDefaults(DefaultParameters);
+        
+        mErrorTolerance = mThisParameters["error_mesh_tolerance"].GetDouble();
+        mConstantError = mThisParameters["error_mesh_constant"].GetDouble();
+        mRemeshingUtilities = ConvertRemeshUtil(mThisParameters["remeshing_utility"].GetString());
+    }
+
+    ///Copy constructor 
+    ErrorMeshCriteria( ErrorMeshCriteria const& rOther )
+      :BaseType(rOther)
+      ,mThisModelPart(rOther.mThisModelPart)
+      ,mDimension(rOther.mDimension)
+      ,mErrorTolerance(rOther.mErrorTolerance)
+      ,mConstantError(rOther.mConstantError)
+    {
+    }
+
+    /// Destructor
+    virtual ~ErrorMeshCriteria() {}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * Compute relative and absolute error.
+     * @param rModelPart Reference to the ModelPart containing the contact problem.
+     * @param rDofSet Reference to the container of the problem's degrees of freedom (stored by the BuilderAndSolver)
+     * @param A System matrix (unused)
+     * @param Dx Vector of results (variations on nodal variables)
+     * @param b RHS vector (residual)
+     * @return true if convergence is achieved, false otherwise
+     */
+
+    bool PostCriteria(
+        ModelPart& rModelPart,
+        DofsArrayType& rDofSet,
+        const TSystemMatrixType& A,
+        const TSystemVectorType& Dx,
+        const TSystemVectorType& b
+    ) override
+    {
+        // We recompute the NODAL_H
+        mFindNodalH.Execute();
+                
+        // We initialize the check
+        bool ConvergedError = true;
+        
+        // We initialize the total error
+        double TotalErrorPow2 = 0.0;
+        double CurrentSolPow2 = 0.0;
+        
+        // Iterate in the nodes
+        NodesArrayType& pNode = rModelPart.Nodes();
+        int numNodes = pNode.end() - pNode.begin();
+        
+//         #pragma omp parallel for 
```

Comment:
you forgot to make it omp parallel? if this is wanted please make a comment and state that it should not be made parallel